Formula/avr-gcc@11.rb

```diff
@@ -177,7 +181,8 @@ def caveats
 
     system "#{bin}/avr-gcc", "-mmcu=atmega328p", "-Os", "-c", "hello.c", "-o", "hello.c.o", "--verbose"
     system "#{bin}/avr-gcc", "hello.c.o", "-o", "hello.c.elf"
-    system "avr-objcopy", "-O", "ihex", "-j", ".text", "-j", ".data", "hello.c.elf", "hello.c.hex"
+    system "#{Formula["avr-binutils"].opt_bin}/avr-objcopy", "-O", "ihex", "-j", ".text", "-j", ".data", \
```

Comment:
why is this needed?

Reply:
I wanted to run the tests that come with this formula in order to have more confidence in my changes, but when I did that, Homebrew printed an error message to the effect that it was unable to run `avr-objcopy`.  I was able to fix it by making this change.  I'm pretty sure that avr-objcopy was not found because it was not on my PATH.  When these tests run, the PATH environment variable is unset, because `ENV.clear` is called at the top of the tests.

(I think the only reason these tests ever worked for anyone was that the shell or some other low-level part of the system probably added some default folders to the PATH, allowing the test script to find the two executables on the PATH that it needs: cat and avr-objcopy.  But on my machine, I didn't have `avr-objcopy` on the PATH.)

I considered puting avr-objcopy onto the PATH at the top of the test.  However, I'm not sure if the AVR binutils will always be on the user's PATH when they are using this toolchain, so I didn't want to make the test environment be like that.

Reply:
Apparently writing `ENV.clear` is not sufficient to prevent the test script from finding various executables on your system like `cat` and `avr-objcopy` and using them.  If you want the tests to run in a more predictable way, the PATH should probably be set to a string that contains a very obscure folder name that will never exist.

Reply:
I used `opt_bin` because that is how we find the AVR binutils in another part of this formula.  I was unable to find information on the differences between `bin` and `opt_bin` and both seem to work fine.

Reply:
> I was unable to find information on the differences between bin and opt_bin and both seem to work fine.

- `opt_bin` --> `/usr/local/opt/foo/bin`
- `bin` --> `/usr/local/Cellar/foo/0.1/bin`

Okay, now I understand the why. But I'm not sure it's needed for everyone, as in your case I'm surprised that there is no avr-objcopy in your $PATH.

avr-binutils must be installed with avr-gcc otherwise the toolchain is completely useless. avr-binutils can be shared by different version of avr-gcc, this is why it's only pulled once the first time.

avr-gcc automatically pulls avr-binutils with the `depends_on "avr-binutils"`, so avr-binutils must be in the path.

This is the normal process:

<img width="898" alt="Screenshot 2021-06-16 at 14 39 36" src="https://user-images.githubusercontent.com/2206544/122220732-e7980e00-ceb0-11eb-806a-8a585beff18e.png">

And this is the test:

<img width="895" alt="Screenshot 2021-06-16 at 14 40 50" src="https://user-images.githubusercontent.com/2206544/122220752-ecf55880-ceb0-11eb-99aa-047d37cb02dc.png">

If you `brew remove` everything osx-cross/avr related and start fresh, what happens?

Reply:
> avr-binutils must be installed with avr-gcc otherwise the toolchain is completely useless.

True, but it does not have to be on the PATH.  GCC is configured with the
`--with-as` and `--with-ld` options to tell it where to find those utilities.
With those two utilities, avr-gcc is able to create an ELF file, and you can
actually use AVRDUDE directly with ELF files instead of converting it to HEX.

OK, I used `brew remove` to remove everything AVR-related.

OK, I did a new test today.  I changed the test in the avr-gcc@11.rb formula
on my computer back to the way it was, so it does not provide the full path
to avr-objcopy during the test.  I have attached a shell session showing:

1. I uninstalled all the software from this repo.
2. I was able to install the avr-gcc@11.rb formula.
3. The toolchain itself works, producing a HEX file that is the
   same as `hello_c_hex` from the test if I manually compile the
   source code from the test.
4. /opt/homebrew/bin/avr-objcopy is actually on the PATH in my shell.
5. The tests still fail when running `avr-objcopy`
   (presumably because the PATH gets unset at the top of the test).
   The error message is pretty vague, but you can see it in the attached
   shell session.

I have to use slightly different commands than you because I'm working with a
formula on my computer instead of on GitHub.

Attachment: [shell_session.txt](https://github.com/osx-cross/homebrew-avr/files/6664614/shell_session.txt)

Reply:
> True, but it does not have to be on the PATH.  GCC is configured with the
> `--with-as` and `--with-ld` options to tell it where to find those utilities.

Ah true! That settles it then!, let's keep the `opt_bin` thing for consistency.